store/db.go

```diff
@@ -49,15 +51,17 @@ func newDBLock(dirPath string) (*dbLock, error) {
 }
 
 func (dl *dbLock) lock() error {
+	dl.Lock()
 	if err := dl.fl.ExclusiveLock(); err != nil {
+		dl.Unlock()
 		return err
 	}
-	dl.Lock()
 	return nil
 }
 
 func (dl *dbLock) unlock() error {
 	if err := dl.fl.Unlock(); err != nil {
+		dl.Unlock()
```

Comment:
So, I went back to the history, found the reason that we do flock() before mutex is because of https://github.com/coreos/rkt/pull/1892#discussion_r48092439